File: node/bucket-listar.js
#!/usr/bin/env node

const { S3Client, ListBucketsCommand } = require("@aws-sdk/client-s3");
const { NodeHttpHandler } = require("@aws-sdk/node-http-handler");
const fs = require('fs');
const { Command } = require('commander');
const program = new Command();

const credentials = JSON.parse(fs.readFileSync('credentials.json'));

const s3Client = new S3Client({
    endpoint: credentials.endpoint_url,
    region: "us-east-1",
    credentials: {
        accessKeyId: credentials.access_key,
        secretAccessKey: credentials.secret_key
    },
    requestHandler: new NodeHttpHandler({
        httpsAgent: new require('https').Agent({
            rejectUnauthorized: false
        })
    }),
    forcePathStyle: true  // Força o estilo de caminho para o endpoint
});

program.parse(process.argv);

const listBuckets = async () => {
    try {
        const data = await s3Client.send(new ListBucketsCommand({}));
        data.Buckets.forEach(bucket => {
            console.log(bucket.Name);
        });
    } catch (err) {
        console.error('Erro ao listar buckets:', err);
    }
};

listBuckets();

File: c#/bucket-listar.cs
using System;
using Amazon.S3;
using Newtonsoft.Json.Linq;
using System.IO;

class ListBuckets
{
    static async Task Main(string[] args)
    {
        var credentials = JObject.Parse(File.ReadAllText("credentials.json"));

        var config = new AmazonS3Config
        {
            ServiceURL = credentials["endpoint_url"].ToString(),
            ForcePathStyle = true
        };

        var s3Client = new AmazonS3Client(credentials["access_key"].ToString(), credentials["secret_key"].ToString(), config);

        try
        {
            var response = await s3Client.ListBucketsAsync();
            foreach (var bucket in response.Buckets)
            {
                Console.WriteLine(bucket.BucketName);
            }
        }
        catch (AmazonS3Exception e)
        {
            Console.WriteLine($"Erro ao listar buckets: {e.Message}");
        }
    }
}
